runtime/src/lib.rs

```diff
@@ -460,6 +457,7 @@ impl_runtime_apis! {
 				None
 			};
 
+			#[allow(clippy::or_fun_call)] // suggestion not helpful here
```

Comment:
Why is it not helpful here? Here is why this lint typically is helpful https://stackoverflow.com/a/45547401/4184410

Reply:
I run into `borrow-checker` issues when applying clippy's suggested change:

```
     Compiling moonbeam-runtime v0.1.0 (/home/stephen/dev/moonbeam/runtime)
  error[E0597]: `config` does not live long enough
     --> /home/stephen/dev/moonbeam/runtime/src/lib.rs:470:5
      |
  470 |                 config.as_ref().unwrap_or_else(<Runtime as pallet_evm::Config>::config),
      |                 ^^^^^^---------
      |                 |
      |                 borrowed value does not live long enough
      |                 argument requires that `config` is borrowed for `'static`
  471 |             ).map_err(|err| err.into())
  472 |         }
      |         - `config` dropped here while still borrowed
```

Reply:
This seems a bit dubious to me. Perhaps this is confusion about what `config` refers to?

Anyway, using a lambda here seems to work, see 9b07d2b

Reply:
I asked about this on the Rust discord. https://discord.com/channels/273534239310479360/274215136414400513/793503318025961524

Maybe we'll get to learn something. Either way, I'm fine with how you have it now.

Reply:
I did update the code to use `unwrap_or_else()`. I just had to use a lambda instead of passing a reference to the `config` function.

Reply:
![image](https://user-images.githubusercontent.com/2915325/103296584-5bf2f980-49c4-11eb-955f-a7d6a0209a90.png)